hub/common_test.go

```
@@ -50,3 +53,16 @@ type failingWriter struct {
 func (f *failingWriter) Write(_ []byte) (int, error) {
 	return 0, f.err
 }
+
+// MustCreateCluster creates a utils.Cluster and calls t.Fatalf() if there is
+// any error.
+func MustCreateCluster(t *testing.T, segs []cluster.SegConfig) *utils.Cluster {
```

why not CreateCluster

---

`MustXXX()` is a marker for developers that the function will handle errors out-of-band (i.e. `panic`) instead of returning them. You typically only want to use a `Must` function in cases where the inputs are hardcoded and known to be correct. Tests are a good use; global initialization is another.

Some examples in the standard library:
- [`regexp.MustCompile`](https://golang.org/pkg/regexp/#MustCompile) vs [`regexp.Compile`](https://golang.org/pkg/regexp/#Compile)
- [`template.Must`](https://golang.org/pkg/text/template/#Must) vs [`template.New`](https://golang.org/pkg/text/template/#Parse)

---

Thanks for the reference from the standard library. I always find those useful.